xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-image/plugin.js

```diff
@@ -100,13 +101,284 @@
+                  for (var i = 0; i < files.length; i++) {
+                    const file = files.item(i);
+                    const attachmentReference = new XWiki.EntityReference(file.name, XWiki.EntityType.ATTACHMENT,
+                      documentReference);
+                    attachments.push(attachmentsStore.create(attachmentReference, file));
+                  }
+                  return attachments;
+                };
+
+                // Open the file picker
+                filePicker.pickLocalFiles({
+                  accept: "image/*",
+                  multiple: false
+                }).then(function (files) {
+                  const attachments = convertFilesToAttachments(files, editor.config.sourceDocument.documentReference);
+
+                  // Cancel the insertion when no image is picked
+                  if (attachments.length === 0) {
+                    return;
+                  }
+
+                  const attachment = attachments[0];
+
+                  const notification = new XWiki.widgets.Notification(
+                    editor.localization.get('xwiki-image.slash.uploadProgress', attachment.name),
+                    'inprogress');
+
+                  // Upload the selected image
+                  const attachmentReference = XWiki.Model.resolve(attachment.id, XWiki.EntityType.ATTACHMENT);
+                  attachmentsStore.upload(attachmentReference, attachment.file).then(() => {
+                    notification.replace(
+                      new XWiki.widgets.Notification(
+                        editor.localization.get('xwiki-image.slash.uploadSuccess', attachment.name),
+                        'done')
+                    );
+                    // Clear the cache so the new image can appear on next usage of the image quick action
+                    attachmentService.clearCache();
+
+                    // Insert the newly uploaded image
+                    imageWidget.insert({
+                      setImageData: {
+                        resourceReference: {
+                          type: "attach",
+                          typed: false,
+                          reference: attachment.id
+                        },
+                        src: CKEDITOR.plugins.xwikiResource.getResourceURL({
+                          type: "attach",
+                          typed: false,
+                          reference: attachment.id
+                        }, editor)
+                      }
+                    });
+                  }).catch(() => {
+                    notification.replace(
+                      new XWiki.widgets.Notification(
+                        editor.localization.get('xwiki-image.slash.uploadError', attachment.name),
+                        'error')
+                    );
+                    return Promise.reject();
+                  });
+                });
+              });
+
+            });
+
+          return "";
+          }
```

Comment:
```suggestion
            return "";
          }
```

The method above could benefit from some auto-format, the indentation if off and some lines are too long for instance.